hbase-common/src/main/java/org/apache/hadoop/hbase/KeyValue.java

```diff
@@ -76,7 +77,7 @@
  * length and actual tag bytes length.
  */
 @InterfaceAudience.Private
-public class KeyValue implements ExtendedCell, Cloneable {
+public class KeyValue implements ExtendedCell, ContiguousCellFormat, Cloneable {
```

Comment:
Could the methods be added to ExtendedCell or is that different concerns?

Reply:
I think we need not add to ExtendedCell. ExtendedCell is for server to create cells. This new interface only the comparator can understand.